src/psd_tools/api/layers.py

```diff
@@ -810,6 +820,28 @@ def descendants(self, include_clip=True):
                 for clip_layer in layer.clip_layers:
                     yield clip_layer
 
+    def find(self, name):
+        """
+        Returns the first layer found for the given layer name
+        
+        :param name:
+        """
+
+        for layer in self.descendants():
+            if layer.name == name:
+                return layer
```

Comment:
Why not reusing the iterator method?

    for layer in self.findall(name):
        return layer